File: apps/tickets/src/contexts/core/domain/repositories/user.repository.ts
import { Result } from 'neverthrow';
import { User } from '../entities/user';
import { UserId } from '../value-objects/user-id';

export abstract class UserRepository {
  abstract save(user: User): Promise<Result<void, Error>>;
  abstract findById(userId: UserId): Promise<Result<User | null, Error>>;
}

File: apps/api/src/contexts/core/domain/repositories/user_repository.py
from abc import ABC, abstractmethod

from returns.result import Result

from src.contexts.core.domain.entities.user import User
from src.contexts.core.domain.value_objects import UserId


class UserRepository(ABC):
    @abstractmethod
    def save(self, user: User) -> Result[None, Exception]:
        pass

    @abstractmethod
    def find_by_id(self, user_id: UserId) -> Result[User | None, Exception]:
        pass
